docs/rules-from-web-poet.rst

```diff
@@ -170,12 +170,18 @@ The :py:func:`web_poet.handle_urls` decorator in this case is indicating that
 the class ``BSTBookPage`` should be used instead of ``BookPage``
 for the domain ``toscrape.com``.
 
-In order to configure the ``scrapy-poet`` overrides automatically
-using these annotations, you can directly interact with `web-poet`_'s
-``default_registry`` (an instance of :py:class:`web_poet.RulesRegistry
-<web_poet.rules.RulesRegistry>`).
+Using the rules in scrapy-poet
+------------------------------
 
-For example:
+``scrapy-poet`` automatically uses the rules defined by the ``@handle_urls``
+annotation by having the default value of the ``SCRAPY_POET_RULES`` setting set to
+:meth:`web_poet.default_registry.get_rules() <web_poet.rules.RulesRegistry.get_rules>`.
+The :py:meth:`get_rules() <web_poet.rules.RulesRegistry.get_rules>` method of the
+``default_registry`` returns a ``List[ApplyRule]``.
+
+However, if you need to add rules from other external packages, make sure to use
+:func:`web_poet.consume_modules <web_poet.rules.consume_modules>` inside your
+``settings.py`` file as shown below.
```

Comment:
I'm still unsure about this "external packages" thing - what are these rules external to. 

Let's say you have a repo with a Scrapy project and page objects

```
my-project-repo/
├── project-pages/
│   ├── project_pages/
│   │   ├── __init__.py
│   │   └── website1.py
│   └── setup.py
├── my-scrapy-project/
│   ├── my_scrapy_project/
│   │   ├── spiders/
│   │   │   └── website1.py
│   │   ├── inline_pages/
│   │   │   └── website2/
│   │   │       ├── product.py
│   │   │       └── product_list.py
│   │   ├── __init__.py
│   │   ├── items.py
│   │   └── middleware.py
│   └── settings.py
└── setup.py
```

and also a separate installable library with page objects:

```
pages-lib/
├── pages_lib/
│   ├── website1/
│   │   ├── product.py
│   │   └── product_list.py
│   └── website2/
│       ├── product.py
│       └── product_list.py
├── setup.py
└── README.rst      
```

All the page objects use `handle_urls` decorator.

Which of them should be considered external? It seems one should use consume_modules for all packages which are not a part of Scrapy project ("external" means "external to the Scrapy project"?)

For modules inside a Scrapy project it is not needed only because default SpiderLoader calls walk_tree for the entries of SPIDER_MODULES setting, which is (by the way) empty by default, and is only set in the default settings.py template. So, I guess there could be cases when even the page objects defined in the project are not loaded - e.g. a case when a spider is executed in a script via CrawlerProcess, and there is no need to define SPIDER_MODULES.

It seems the safest would be to ask users to call consume_modules for all packages which may contain page objects, and either clarify "external" vs "internal" terminology, or stop using it. I'd also not go into details explaining SPIDER_MODULES in our docs.

It may also be not a bad idea to add PAGE_OBJECT_PACKAGES (or PAGE_OBJECT_MODULES) option. If it's easier to add this option than to fix the documentaion, it makes sense to do it in this PR; otherwise, it's better to do it separately.

Reply:
I see what you mean. Thanks for laying it down.

> It seems the safest would be to ask users to call consume\_modules for all packages which may contain page objects

Yeah, it would seem that would ensure it would work for all cases.

> It may also be not a bad idea to add PAGE\_OBJECT\_PACKAGES (or PAGE\_OBJECT\_MODULES - not precise, but more similar to SPIDER\_MODULES) option

That's actually much easier to explain. It's worth going this route than directly exposing `consume_modules`.:+1: 

> It also might be a good idea to separate the CI fixes to a different PR, though not required.

Good idea. I've moved it here: #125

Reply:
What do you think about the new `SCRAPY_POET_MODULES` implemented in af2cace ?